doc/Design.md

```diff
@@ -143,3 +159,142 @@ dereferencing. Unlike shared buffers, which can only be a buffer type in a
 capsule, granted memory can be defined as any type. Therefore, processes cannot
 access this memory since doing so might violate type-safety.
 
+## In-Kernel Design Principles
+
+To help meet Tock's goals, encourage portability across hardware, and ensure a
+sustainable operating system, several design principles have emerged over time
+for the Tock kernel. These are general principles that new contributions to the
+kernel should try to uphold. However, these principles have been informed by
+Tock's development, and will likely continue to evolve as Tock and the Rust
+ecosystem evolve.
+
+### Role of HILs
+
+Generally, the Tock kernel is structured into three layers:
+
+1. Chip-specific drivers: these typically live in a crate in the
+   `chips` subdirectory, or an equivalent crate in an different repository
+   (e.g. the Titan port is out of tree but its `h1b` create is the
+   equivalent here). These drivers have implementations that are specific
+   to the hardware of a particular microcontroller. Ideally,
+   their implementation is fairly simple, and they merely adhere to a
+   common interface (a HIL). That's not always the case, but that's
+   the ideal.
+
+2. Chip-agnostic, portable, peripheral drivers and subsystems. These
+   typically live in the `capsules` crate. These includes things like
+   the virtual alarms and virtual I2C stack, as well as drivers for
+   hardware peripherals not on the chip itself (e.g. sensors, radios,
+   etc). These drivers typically rely on the chip-specific drivers
+   through the HILs.
+
+3. System call drivers, also typically found in the `capsules`
+   crate. These are the drivers that implement a particular part of
+   the system call interfaces, and are often even more abstracted from
+   the hardware than (2) - for example, the temperature sensor system
+   call driver can use any temperature sensor, including several
+   implemented as portable peripheral drivers.
+
+
+   The system call interface is another point of
+   standardization that can be implemented in various ways. So it’s
+   perfectly reasonable to have several implementations of the same
+   system call interface that use completely different hardware
+   stacks, and therefore HILs and chip-specific drivers (e.g. a
+   console driver that operates over USB might just be implemented as
+   a different system call driver that implements the same system
+   calls, rather than trying to fit USB into the UART HIL).
+
+The connective tissue between layers are the HILs. The HIL interfaces
```

Comment:
This is a bad metaphor, because information does not flow through connective tissue. Use a technical term.